src/components/ResourceBrowser/ResourceBrowser.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useMemo } from 'react'
+import { NavLink, useHistory } from 'react-router-dom'
+import { showError, getUserRole, DevtronProgressing, useAsync } from '@devtron-labs/devtron-fe-common-lib'
+import PageHeader from '../common/header/PageHeader'
+import { sortObjectArrayAlphabetically } from '../common'
+import { ALL_NAMESPACE_OPTION, K8S_EMPTY_GROUP, SIDEBAR_KEYS } from './Constants'
+import { URLS } from '../../config'
+import ClusterSelectionList from '../ClusterNodes/ClusterSelectionList'
+import { getClusterList } from '../ClusterNodes/clusterNodes.service'
+import { ClusterDetail } from '../ClusterNodes/types'
+import { ReactComponent as Add } from '../../assets/icons/ic-add.svg'
+import './ResourceBrowser.scss'
+
+/* NOTE: anonymous funcitons are references
+ * when we export default we need to pass a reference or value
+ * Thus not naming function */
+const ResourceBrowser: React.FC<Record<string, never>> = () => {
+    const { push } = useHistory()
+
+    /* this list is to be used by ClusterSelectionList to refresh on demand;
+     * thus not immediately fetching results, it will only run when reloadClusterList is called */
+    const [clusterListLoading, clusterList, clusterListError, reloadClusterList] = useAsync(getClusterList, [], false)
+    const [initialLoading, data, error] = useAsync(() =>
+        Promise.all([getClusterList(), window._env_.K8S_CLIENT ? null : getUserRole()]),
```

Comment:
Can we combine above two services and can we use Promise.allSettled

Reply:
This would not be possible since I need two different loading states. One for the initial page load & one for the shimmer effect on the list elements when we fresh the refresh button.